zebra-rpc/src/methods/get_block_template_rpcs/types/get_block_template.rs

```diff
@@ -234,3 +234,60 @@ impl GetBlockTemplate {
         }
     }
 }
+
+/// Error response to a `getblocktemplate` RPC request in proposal mode.
+#[derive(Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum ProposalRejectReason {
+    /// Block rejected as invalid
+    Rejected,
+}
+
+/// Response to a `getblocktemplate` RPC request in proposal mode.
+#[derive(Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+#[serde(untagged, rename_all = "kebab-case")]
+pub enum ProposalResponse {
+    /// Block was not successfully submitted, return error
+    ErrorResponse {
+        /// Reason the proposal was invalid as-is
+        reject_reason: ProposalRejectReason,
+
+        /// The getblocktemplate RPC capabilities supported by Zebra.
+        capabilities: Vec<String>,
+    },
+
+    /// Block successfully proposed, returns null
+    Valid,
+}
+
+#[derive(Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+#[serde(untagged)]
+/// A `getblocktemplate` RPC response.
+pub enum Response {
+    /// `getblocktemplate` RPC request in template mode.
+    TemplateMode(Box<GetBlockTemplate>),
+
+    /// `getblocktemplate` RPC request in proposal mode.
+    ProposalMode(ProposalResponse),
+}
+
+impl From<ProposalRejectReason> for ProposalResponse {
+    fn from(reject_reason: ProposalRejectReason) -> Self {
+        // Convert default values
+        let capabilities: Vec<String> = GET_BLOCK_TEMPLATE_CAPABILITIES_FIELD
+            .iter()
+            .map(ToString::to_string)
+            .collect();
```

Comment:
Optional: what happens if Zebra is configured in a way that changes the default capabilities?

Reply:
Moved to a fn so it stays consistent in [`31472a8`](https://github.com/ZcashFoundation/zebra/pull/5870/commits/31472a86b1d0da1ff6f6a9ea577fb52b04017b81).